platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -0,0 +1,626 @@
+    }
+
+    return OTA_PAL_COMBINE_ERR( mainErr, subErr );
+}
+
+/* Verify the signature of the input content with OpenSSL. */
+static OtaPalMainStatus_t Openssl_DigestVerify( EVP_MD_CTX * pSigContext,
+                                                EVP_PKEY * pPkey,
+                                                FILE * pFile,
+                                                Sig256_t * pSignature )
+{
+    OtaPalMainStatus_t mainErr = OtaPalSuccess;
+    size_t bytesRead;
+    uint8_t * pBuf;
+
+    /* Verify an ECDSA-SHA256 signature. */
+    if( ( pSigContext != NULL ) &&
+        ( pPkey != NULL ) &&
+        ( pFile != NULL ) &&
+        ( 1 == EVP_DigestVerifyInit( pSigContext, NULL, EVP_sha256(), NULL, pPkey ) ) )
+    {
+        LogDebug( ( "Started signature verification." ) );
+
+        pBuf = OPENSSL_malloc( OTA_PAL_LINUX_BUF_SIZE );
+
+        if( pBuf != NULL )
+        {
+            /* Rewind the received file to the beginning. */
+            /* Linux port using standard library */
+            /* coverity[misra_c_2012_rule_21_6_violation] */
+            if( fseek( pFile, 0L, SEEK_SET ) == 0 ) /*lint !e586
+                                                     * C standard library call is being used for portability. */
+            {
+                do
+                {
+                    /* Linux port using standard library */
+                    /* coverity[misra_c_2012_rule_21_6_violation] */
+                    bytesRead = fread( pBuf, 1U, OTA_PAL_LINUX_BUF_SIZE, pFile ); /*lint !e586
+                                                                                   * C standard library call is being used for portability. */
+                    /* Include the file chunk in the signature validation. Zero size is OK. */
+                    ( void ) EVP_DigestVerifyUpdate( pSigContext, pBuf, bytesRead );
+                } while( bytesRead > 0UL );
+
+                if( 1 != EVP_DigestVerifyFinal( pSigContext,
+                                                pSignature->data,
+                                                pSignature->size ) ) /*lint !e732 !e9034 Allow comparison in this context. */
+                {
+                    LogError( ( "File signature check failed at FINAL" ) );
+                    mainErr = OtaPalSignatureCheckFailed;
+                }
+            }
+
+            /* Free the temporary file page buffer. */
+            OPENSSL_free( pBuf );
+        }
+        else
+        {
+            LogError( ( "Failed to allocate buffer memory." ) );
+            mainErr = OtaPalOutOfMemory;
+        }
+    }
+    else
+    {
+        LogError( ( "File signature check failed at INIT." ) );
+        mainErr = OtaPalSignatureCheckFailed;
+    }
+
+    return mainErr;
+}
+
+/* Verify the signature of the specified file using OpenSSL. */
+static OtaPalStatus_t otaPal_CheckFileSignature( OtaFileContext_t * const C )
+{
+    OtaPalMainStatus_t mainErr = OtaPalSuccess;
```

Comment:
Initialize return status to failure (see previous comment for the reason):

```c
OtaPalMainStatus_t mainErr = OtaPalSignatureCheckFailed; 
```

Then just delete line 440.